### ISSUE TITLE
vitest will crash when using ffi-napi

### ISSUE THREAD

**Issue by LizardLiang** [vitest-dev/vitest#2091]
### Describe the bug

when using `ffi-napi` module, vitest cannot fully finish its job
![image](https://user-images.githubusercontent.com/12999887/192684590-a98d6b12-7eff-48fa-928a-420ceb1f5dd0.png)

after i remove file include ffi-napi, it becomes normal again
![image](https://user-images.githubusercontent.com/12999887/192684820-49a3cd22-0d9f-48b2-8b78-e799d474b3bb.png)

### Reproduction

In any testing file  add
`const  ffi = require('ffi-napi')`

### System Info

```shell
System:
    OS: Windows 10 10.0.19043
    CPU: (16) x64 Intel(R) Core(TM) i7-10700 CPU @ 2.90GHz
    Memory: 16.55 GB / 31.80 GB
  Binaries:
    Node: 16.17.0 - ~\scoop\apps\nvm\current\nodejs\nodejs\node.EXE
    npm: 8.15.0 - ~\scoop\apps\nvm\current\nodejs\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (105.0.1343.50)
    Internet Explorer: 11.0.19041.1566
  npmPackages:
    @vitejs/plugin-react: ^2.1.0 => 2.1.0
    vite: ^3.1.0 => 3.1.3
    vitest: ^0.23.4 => 0.23.4
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [ ] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sheremet-va** [comment]
According to https://github.com/node-ffi-napi/node-ffi-napi/issues/125 you cannot use it inside a worker thread. Vitest runs tests only inside worker threads.

**Comment by marcoroth** [comment]
Is there any workaround for this? Or do I have to ditch vitest when testing modules using `ffi-napi`? 🥲

**Comment by AriPerkkio** [comment]
> Is there any workaround for this? 

Use `--no-threads`. Or if you are using v1-beta releases, use `--pool=forks`.

**Comment by AriPerkkio** [comment]
Fixed by https://github.com/vitest-dev/vitest/pull/3925 and https://github.com/vitest-dev/vitest/pull/4172.

Vitest 1.0.0-beta has now `--pool=forks` which uses multiple `node:child_process` in parallel. The previous `--no-threads` was using a single `node:child_process`.

If the code you are testing is incompatible with `node:worker_threads`, switch to `--pool=forks`. If you run into same error with that pool, feel free to open new issue with minimal reproduction.

**Comment by marcoroth** [comment]
Awesome, thanks @AriPerkkio! Using `--pool=forks` works perfectly!
